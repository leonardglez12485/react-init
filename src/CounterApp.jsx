import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {
    const [counter, setCounter] = useState( value )
    const handleAdd = ()=>{
       setCounter((c)=> c + 1);
    }

    const handleSustract = ()=>{
       setCounter((c)=> c - 1);
    }

    const handleReset = ()=>{
       setCounter(value);
    }

  return (
    <>
       <h1>CounterApp</h1>
       <h2> { counter } </h2>
       <button onClick={handleAdd}>
       +
       </button>
       <button onClick={handleSustract}>
       -
       </button>
       <button onClick={handleReset}>
       Reset
       </button>
    </>
    
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};



