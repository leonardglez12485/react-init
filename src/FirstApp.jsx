
import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {
    if(!title) throw new Error('Title not exist')
  return (
    <>
     <h2>{ title}</h2>
     <p>{subtitle}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
 title: 'No title',
  subtitle: 'No subtitle',
}
