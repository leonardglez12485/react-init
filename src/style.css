html, body{
    color: white;
    background-color: #21232A;
    font-family: Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}

button{
    background-color: #2C2F33;
    color: white;
    border: none;
    padding: 10px 20px;
    font-size: 1.3rem;
    cursor: pointer;
    margin-right: 5px;
}