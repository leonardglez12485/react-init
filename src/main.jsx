import React from 'react';
import ReactDOM from 'react-dom/client';
//import { HelloWorldApp } from './HelloWorldApp';
//import { FirstApp } from './FirstApp';
import './style.css'
import { CounterApp } from './CounterApp';




ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
      {/*<HelloWorldApp/>*/}
     <CounterApp value={25}/>
{  /*  <FirstApp title=" Hola Pepe" subtitle="Hi" name="Leonard"/> */}    </React.StrictMode>
)
