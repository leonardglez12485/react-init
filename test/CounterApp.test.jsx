import { fireEvent, render, screen } from '@testing-library/react'
import { CounterApp } from '../src/CounterApp'

describe('CounterApp Componet', ()=>{

    const value = 25;

    test('should be match whit snapshot',()=>{
        const {container} =render(<CounterApp value={value}/>)
        expect(container).toMatchSnapshot();
    })

    test('should be display value = 25', ()=>{
        render(<CounterApp value={value}/>)
        expect(Number(screen.getByRole('heading',{level:2}).innerHTML)).toEqual(25);
    })

    test('should be inceres 1 whith the button', ()=>{
        render(<CounterApp value={value}/>)
        fireEvent.click( screen.getByText('+'))
        expect(Number(screen.getByRole('heading',{level:2}).innerHTML)).toEqual(26);
    })

    test('should be decres 1 whith the button', ()=>{
        render(<CounterApp value={value}/>)
        fireEvent.click( screen.getByText('-'))
        expect(Number(screen.getByRole('heading',{level:2}).innerHTML)).toEqual(24);
    })

})