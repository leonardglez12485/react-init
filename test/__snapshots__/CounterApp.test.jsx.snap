// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Componet should be match whit snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    25
     
  </h2>
  <button>
    +
  </button>
  <button>
    -
  </button>
  <button>
    Reset
  </button>
</div>
`;
